packages/liquidator/src/liquidationStrategy.js

```diff
@@ -115,17 +115,29 @@ module.exports = (
   }
 
   // Call this with only undercollateralized position
-  // SyntheticTokenBalance should be passed in respecting maxTokensToLiquidate
+  // SyntheticTokenBalance should be passed in respecting maxTokensToLiquidate.
+  // Responsible for emitting customized logs since its already parsing
+  // whether a liquidation is valid to send and for how many tokens.
   function processPosition({
     position,
     syntheticTokenBalance,
     currentBlockTime,
     maxCollateralPerToken,
     // mandated token to liquidate amount, an override
     maxTokensToLiquidateWei,
+    shouldEmitWDFSkipLogs,
     ...logInfo
   }) {
-    assert(syntheticTokenBalance, "requires syntheticTokenBalance");
+    if (!syntheticTokenBalance || toBN(syntheticTokenBalance).lte(toBN(0))) {
```

Comment:
im ok with this still throwing an error rather than returning. wdyt?

Reply:
Yeah I actually changed this to make an existing test that checks for an `error` log pass, but I'll just remove that test. I'd rather reduce bot churn

Reply:
yea np. i actually meant that it could both log and throw error. but i see you changed it to just throw error, which i am fine with too.